kobo-uncaged/kunc/kunc.go

```diff
@@ -204,31 +199,35 @@ func (ku *koboUncaged) GetBook(book uc.BookID, filePos int64) (io.ReadCloser, in
 	bkPath := util.ContentIDtoBkPath(ku.k.BKRootDir, cid, string(ku.k.ContentIDprefix))
 	fi, err := os.Stat(bkPath)
 	if err != nil {
-		return nil, 0, errors.Wrap(err, "error getting book stats")
+		return nil, 0, fmt.Errorf("GetBook: error getting book stats: %w", err)
 	}
 	bookLen := fi.Size()
 	ebook, err := os.OpenFile(bkPath, os.O_RDONLY, 0644)
+	if err != nil {
+		err = fmt.Errorf("GetBook: error opening book file: %w", err)
+	}
 	return ebook, bookLen, err
 }
 
 // DeleteBook instructs the client to delete the specified book on the device
 // Error is returned if the book was unable to be deleted
 func (ku *koboUncaged) DeleteBook(book uc.BookID) error {
+	var err error
 	// Start with basic book deletion. A more fancy implementation can come later
 	// (eg: removing cover image remnants etc)
 	cid := util.LpathToContentID(book.Lpath, string(ku.k.ContentIDprefix))
 	bkPath := util.ContentIDtoBkPath(ku.k.BKRootDir, cid, string(ku.k.ContentIDprefix))
 	dir, _ := filepath.Split(bkPath)
 	dirPath := filepath.Clean(dir)
-	err := os.Remove(bkPath)
-	if err != nil {
-		return errors.Wrap(err, "error deleting book")
+	if ku.k.KuConfig.EnableDebug {
+		log.Printf("[DEBUG] CID: %s, bkPath: %s, dir: %s, dirPath: %s\n", cid, bkPath, dir, dirPath)
+	}
+	if err = os.Remove(bkPath); err != nil {
```

Comment:
I would probably ignore the error if it's due to a file not existing.
```suggestion
	if err = os.Remove(bkPath); err != nil && !os.IsNotExist(err) {
```

Reply:
Good idea.